00-Raster-Structure.Rmd

```diff
@@ -353,44 +346,107 @@ GDALinfo("NEON_RemoteSensing/HARV/DSM/HARV_dsmCrop.tif")
+
+However, raster data can also be multi-band meaning that within one file it
+contains data on multiple variables for each cell. A raster that is a color 
+photo of a landsape is a good example of multi-band rasters.  The true colors we 
+percieve in a landscape consists of many different wavelengths of light, when we
+digitalize those they tend to fall into three categories: red, green, and blue. 
+Each of these colors becomes a band and the combination of all three bands within
+each cell gives the true color map image.  
+
+By default the `raster` function only imports the first band in a raster. 
+Multi-band rasters in Lesson 04:  
+
+<a href="{{ site.baseurl }}/NEON-R-Spatial-Raster/R/Multi-Band-Rasters-In-R/"  target="_blank"> Working with Multi-band Rasters: Images in R</a>.
+
+#COLIN ![Image on what bands are...]()
+
+##Viewing Raster Attributes Prior to Reading in a File
+As mentioned above, a GeoTIFF contains a set of embedded tags that contain 
+metadata about the file. So far we've viewed attribute data after reading in a 
+raster file.  However, we could also view those tags before we open the data in 
+`R` using the `GDALinfo()` function.
+
+Using the same file DSM file we've already used for illustration: 
+```{r view-attributes-gdal}
+# view attributes before opening file
+GDALinfo("NEON_RemoteSensing/HARV/DSM/HARV_dsmCrop.tif")
+
+```
 
-It is important to note that rasters can be either single or multi-band. If you 
-remember, GDALinfo told us the `DSM` raster that we are working with only has
-one band. 
+Notice a few things in the output:
 
-By default the `raster` function only imports the first band in a raster so we can
-use it for our `DSM`. We will explore multi-band rasters in a [later lesson 4??]()
+1. A projection is described as a string - this format is called `proj4`
+   `+proj=utm +zone=18 +datum=WGS84 +units=m +no_defs `
+2. We can se a `NoDataValue` (-9999)
+3. We can tell how many `bands` the file contains (1)
+4. We can view the x and y `resolution` of the data (1)
+5. We can see the min and max values of the data `Bmin` and `Bmax`
 
-#COLIN
-![Image on what bands are...]()
+This is great - we haven't yet opened the file but we already know some things
+about it.
 
+What else do you see in the attributes of our raster? Take note as we will 
+explore these metadata later in the lesson.
 
-#Challenge
+#Challenge: Viewing Raster Attributes 
+In Lesson 01 we will be using the file NEON_RemoteSensing/HARV/DSM/HARV_DSMhill.tif
+Before we read in the file let's learn a bit about it's attributes.  Without
+reading in the file can you determine: 
+1. If this file has the same CRS as DSM?
+2. What format NoData values take?
+3. The resolution of the raster data? 
+4. How large a 5x5 pixel area would be? 
+5. If the file is a multi- or single-band raster?
+
+``` {r challenge-code}
+GDALinfo("NEON_RemoteSensing/HARV/DSM/HARV_DSMhill.tif")
+```
 
-???
+Checking back in
```

Comment:
This is both review and answers to the challenge above.  I still can't figure out how it would work in the rendered page but I want to provide the code/ answers to the challenges in some format or another so that people can check their thinking.  At the same time I don't like having it immediately following the challenge as that doesn't make them work.  We can discuss later how to link out or move to end etc, but for now I've just been adding the challenge answer directly after the challenge.  

Ending each lesson with a review/challenge isn't a bad idea.